activesupport/lib/active_support/file_evented_update_checker.rb

```diff
@@ -0,0 +1,72 @@
+require 'listen'
+
+module ActiveSupport
+  class FileEventedUpdateChecker
+    attr_reader :listener
+    def initialize(files, directories={}, &block)
+      @files = Set.new
```

Comment:
But @files is of Set type so duplicate entries will be discarded and apart from that @files contains absolute path. So because of these reasons I have kept this instance variable.